include/classify/multiclass_dataset.h

```diff
@@ -0,0 +1,189 @@
+/**
+ * @file multiclass_dataset.h
+ * @author Chase Geigle
+ *
+ * All files in META are released under the MIT license. For more details,
+ * consult the file LICENSE in the root of the project.
+ */
+
+#ifndef META_CLASSIFY_MULTICLASS_DATASET_H_
+#define META_CLASSIFY_MULTICLASS_DATASET_H_
+
+#include "learn/dataset.h"
+#include "util/invertible_map.h"
+
+namespace meta
+{
+namespace classify
+{
+
+class multiclass_dataset : public learn::labeled_dataset<class_label>
+{
+  public:
+    using class_label_map = util::invertible_map<label_type, label_id>;
+    using class_label_iterator = class_label_map::iterator;
+
+    /**
+     * Creates an in-memory dataset from a forward_index. This loads the
+     * **entire index** into memory, so you should only use this on small
+     * datasets.
+     */
```

Comment:
So you'd still use this *class*, but not that specific *constructor*.

Fixed in 7c81b64.